docs/quickstart/regression.md

```diff
@@ -0,0 +1,122 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.1
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# Regression
+
++++
+
+Regression analysis is a set of statistical methodologies for determining the relationship between a dependent (or outcome) variable and one or more independent variables (also known as predictor variables). The most common algorithm used is Linear Regression. 
+
+Linear Regression is a supervised learning technique that attempts to fit the best line that reduces the discrepancies between the predicted and the actual target value. This best fit line is arrived at by aiming to reduce the MSE (Mean Square Error).
+
++++
+
+Let's import the required libraries and read the dataset.
+
+```{code-cell} ipython3
+import urllib.request
+import pandas as pd 
+import seaborn as sns
+import warnings
+warnings.filterwarnings('ignore')
+from sklearn.preprocessing import LabelEncoder
+from sklearn.linear_model import LinearRegression
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import r2_score
+from sklearn_evaluation import plot
+
+# download dataset
+urllib.request.urlretrieve('https://raw.githubusercontent.com/ploomber/sklearn-evaluation/master/docs-assets/regression/insurance.csv', filename='insurance.csv')
+
+data = pd.read_csv('insurance.csv')
+```
+
+## Analyse the dataset
+
+```{code-cell} ipython3
+data.head()
+```
+
+Transform non-numerical labels to numerical labels.
+
+```{code-cell} ipython3
+#sex
+le = LabelEncoder()
+le.fit(data.sex.drop_duplicates()) 
+data.sex = le.transform(data.sex)
+# smoker or not
+le.fit(data.smoker.drop_duplicates()) 
+data.smoker = le.transform(data.smoker)
+#region
+le.fit(data.region.drop_duplicates()) 
+data.region = le.transform(data.region)
+```
+
+Let's visualise the correlation among the variables.
+
+```{code-cell} ipython3
+sns.heatmap(data.corr(), annot=True,cmap='cool')
+```
+
+A strong correlation is observed with the smoking aspect of the patient. Now let's train a linear regression model on the data.
+
++++
+
+## Train the model
+
+```{code-cell} ipython3
+X = data.drop(['charges'], axis = 1)
+y = data.charges
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
+model = LinearRegression().fit(X_train,y_train)
+```
+
+```{code-cell} ipython3
+y_pred = model.predict(X_test)
+print(r2_score(y_test, y_pred))
+```
+
+## Residual Plot
+
++++
+
+R2 score is one measure for measuring how well the model fits the data. But we need other approaches as well to analyse the results. 
```

Comment:
`one measure for measuring` sounds redundant. maybe `R2 score measures how well...` try passing this through Grammarly to spot this problems

Reply:
The free version suggested some minor changes, I have incroporated those.